### PR TITLE
feat: honor Caps Lock status when autocompletion.

### PR DESCRIPTION
**Pull request by scnjdl**
**Summary**
Input lowercase or uppercase keyword per user input

**Testing**
manual testing

### CODE REVIEW

**Comment by scnjdl** [comment]
The code is updated to make the completion list refresh for each typing.

File: server/src/sas/CompletionProvider.ts [hunk 1]
```diff
@@ -265,26 +280,39 @@ export class CompletionProvider {
     }
   }
 
-  getCompleteItems(position: Position): Promise<CompletionItem[] | undefined> {
+  getCompleteItems(
+    position: Position,
+  ): Promise<CompletionItem[] | CompletionList | undefined> {
     return new Promise((resolve) => {
       this._getZone(position);
+      const prefix = this._getPrefix(position);
```
Comment (scnwwu) [inline server/src/sas/CompletionProvider.ts:288]: nit: this.popupContext.prefix is already there, no need to calculate again

**Review by scnwwu — COMMENTED** [review]
How about changing the title to be something like
follow user's typing case
as we're not really checking Caps Lock status

File: server/src/sas/CompletionProvider.ts (hunk 1 above)
Comment (scnjdl) [reply server/src/sas/CompletionProvider.ts]: Done

**Review by scnwwu — APPROVED** [review]

**Comment by liu-lihua** [comment]
Assign it to @Zhirong2022 for interactive testing since it is ready for testing now.

**Review by Zhirong2022 — APPROVED** [review]

**Pull request merged**